File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> mapp;
        int sum = 0;
        int count = 0;
        mapp[0] = 1;

        for(int i = 0;i < nums.size(); i++){
            sum += nums[i];
            int remain = sum - k;
            if(mapp.find(remain) != mapp.end()){
                count+=mapp[remain];
            }
            mapp[sum]++;
        }
        return count;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        mapp = defaultdict(int)
        summ = 0
        mapp[0] = 1
        ans = 0

        for i in range(len(nums)):
            summ += nums[i]

            remain = summ - k
            if remain in mapp:
                ans += mapp[remain]
            
            mapp[summ] += 1
        
        return ans